superset/views.py

```diff
@@ -2199,10 +2199,17 @@ def sqllab_viz(self):
                 dims.append(col)
             agg = config.get('agg')
             if agg:
-                metrics.append(models.SqlMetric(
-                    metric_name="{agg}__{column_name}".format(**locals()),
-                    expression="{agg}({column_name})".format(**locals()),
-                ))
+                if agg == 'count_distinct':
```

Comment:
I agree with @bkyryliuk that both codepaths should ideally rely on the same logic, but I'm the one who originally took the shortcut here and I'd be ok to keep this refactor out of the scope of this PR. Say the scope of this PR is to fix the bug, another PR can refactor the metric creation logic. 

I'll have a much bigger refactor changing the way we deal with metrics altogether...